lib/experimental/blocks.php

```diff
@@ -106,8 +98,18 @@ function wp_enqueue_block_view_script( $block_name, $args ) {
 		 * @param WP_Block $block_instance The block instance.
 		 */
 		function process_block_bindings( $block_content, $block, $block_instance ) {
-			// If the block doesn't have the bindings property, return.
-			if ( ! isset( $block['attrs']['metadata']['bindings'] ) ) {
+
+			// Allowed blocks that support block bindings.
+			// TODO: Look for a mechanism to opt-in for this. Maybe adding a property to block attributes?
+			$allowed_blocks = array(
```

Comment:
We have a similar check in `block-supports/pattern.php` as seen here: https://github.com/WordPress/gutenberg/pull/57789/files#r1452055620. Do you think we should duplicate this array there? Or should we move this into a sharable JSON config so that it can also be imported by JS? Not sure if it's worth the effort though since it's supposed to be a temporary solution.

Reply:
I think question number one is whether these two lists should always match. Eventually, it should be possible to use block bindings with all blocks and I expect to see a method to opt in for that. The prerequisite is the full support in HTML API to replace HTML tags in the saved content.

Reply:
Yeah, I was basically tossing out three different solutions above 😅:

1. Duplicate the `$allowed_blocks` config in `block-supports/pattern.php`.
2. Move the constant to a sharable PHP file that can be imported by both PHP scripts.
3. Go a bit further and move the constant to a sharable JSON file that can also be imported by client-side JS.

Reply:
I think duplicating it is ok for now.

I've put together a PR that does that - [Update pattern overrides to use a hard coded support array](https://github.com/WordPress/gutenberg/pull/57912#top)

It should ensure pattern overrides don't break when this PR is merged (I think this PR is currently a little bit behind trunk).